lookout/style/format/quality_report.py

```diff
@@ -1,384 +1,321 @@
 
-from lookout.style.format.analyzer import FormatAnalyzer
+from lookout.style.format.analyzer import FixData, FormatAnalyzer
 from lookout.style.format.benchmarks.time_profile import profile
+from lookout.style.format.classes import CLASS_REPRESENTATIONS
+from lookout.style.format.descriptions import describe_rule
 from lookout.style.format.feature_extractor import FeatureExtractor
 from lookout.style.format.model import FormatModel
-from lookout.style.format.postprocess import filter_uast_breaking_preds
 from lookout.style.format.utils import generate_comment, merge_dicts, prepare_files
 from lookout.style.format.virtual_node import VirtualNode
 
 
-def generate_report(y, y_pred, vnodes_y, n_files, target_names):
-    """Generate report: classification report, confusion matrix, files with most errors."""
-    with io.StringIO() as output:
-        print("Classification report:\n" + classification_report(y, y_pred,
-                                                                 target_names=target_names),
-              file=output)
-        print("Confusion matrix:\n" + str(confusion_matrix(y, y_pred)), file=output)
-
-        # sort files by mispredictions and print them
-        file_mispred = []
-        for gt, pred, vn in zip(y, y_pred, vnodes_y):
-            if gt != pred:
-                file_mispred.append(vn.path)
-        file_stat = Counter(file_mispred)
+def convert_to_sklearn_format(vnodes: Iterable[VirtualNode]
+                              ) -> Tuple[List[int], List[int], List[VirtualNode], List[str]]:
+    """
+    Convert VirtualNode-s sequence to true targets, predicted targets and corresponding vnodes.
 
-        to_show = file_stat.most_common()
-        if n_files > 0:
-            to_show = to_show[:n_files]
+    :param vnodes: VirtualNode-s sequence.
+    :return: true targets, predicted targets, corresponding vnodes and target names.
+    """
+    y, y_pred, vnodes_y = [], [], []
+    class_sequences_to_labels = defaultdict(lambda: len(class_sequences_to_labels))
+    for vnode in vnodes:
+        if vnode.y is None:
+            continue
+        vnodes_y.append(vnode)
+        if hasattr(vnode, "y_old"):
+            y.append(class_sequences_to_labels[vnode.y_old])
+            y_pred.append(class_sequences_to_labels[vnode.y])
+        else:
+            y.append(class_sequences_to_labels[vnode.y])
+            y_pred.append(class_sequences_to_labels[vnode.y])
+
+    labels_to_class_sequences = {label: seq for seq, label in class_sequences_to_labels.items()}
+    target_names = ["".join(CLASS_REPRESENTATIONS[cls] for cls in labels_to_class_sequences[i])
+                    for i in range(len(labels_to_class_sequences))]
+
+    return y, y_pred, vnodes_y, target_names
+
+
+def generate_report(vnodes, n_files):
+    """Generate report: classification report, confusion matrix, files with most errors."""
+    y, y_pred, vnodes_y, target_names = convert_to_sklearn_format(vnodes)
+    # classification report
```

Comment:
Please remove empty new liens in this function because it is a nightmare to review otherwise